doc: update environment helpers in dockerTools docs, add fakeNss section (#285223)
NixOS/nixpkgs: doc/build-helpers/special/fakenss.section.md

# fakeNss {#sec-fakeNss}

Provides `/etc/passwd` and `/etc/group` files that contain `root` and `nobody`, allowing user/group lookups to work in binaries that insist on doing those.
This might be a better choice than a custom script running `useradd` and related utilities if you only need those files to exist with some entries.

`fakeNss` also provides `/etc/nsswitch.conf`, configuring NSS host resolution to first check `/etc/hosts` before checking DNS, since the default in the absence of a config file (`dns [!UNAVAIL=return] files`) is quite unexpected.

> It also creates an empty directory at `/var/empty` because it uses that as the home directory for the `root` and `nobody` users.

Reviewer: I don't think `root` should have its home there, because that makes it all too easy to accidentally put a file there during image creation.

Author: This seems like a change to make in the code rather than the docs. For now I could add a note to be careful of this detail, what do you think?

> The `/var/empty` directory can also be used for binaries that use that directory as a `chroot` target to prevent file access in processes that do not need to access files.

Reviewer: ```suggestion
The `/var/empty` directory can also be used as a `chroot` target to prevent file access in processes that do not need to access files, if your container runs such processes.
```
